pkg/reconciler/kubernetes/tektoninstallerset/client/check.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/reconciler/shared/hash"
+	"go.uber.org/zap"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"knative.dev/pkg/logging"
+)
+
+func (i *InstallerSetClient) CheckSet(ctx context.Context, comp v1alpha1.TektonComponent, isType string) ([]v1alpha1.TektonInstallerSet, error) {
+	logger := logging.FromContext(ctx)
+
+	labelSelector := labels.NewSelector()
+	createdReq, _ := labels.NewRequirement(v1alpha1.CreatedByKey, selection.Equals, []string{i.resourceKind})
+	if createdReq != nil {
+		labelSelector = labelSelector.Add(*createdReq)
+	}
+	typeReq, _ := labels.NewRequirement(v1alpha1.InstallerSetType, selection.Equals, []string{isType})
+	if typeReq != nil {
+		labelSelector = labelSelector.Add(*typeReq)
+	}
+
+	logger.Infof("%v/%v: checking installer sets with labels: %v", i.resourceKind, isType, labelSelector.String())
+
+	is, err := i.clientSet.List(ctx, v1.ListOptions{LabelSelector: labelSelector.String()})
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Infof("%v/%v: found %v installer sets", i.resourceKind, isType, len(is.Items))
+
+	iSets := is.Items
+
+	if len(iSets) == 0 {
+		logger.Infof("%v/%v: installer sets not found", i.resourceKind, isType)
+		return nil, ErrNotFound
+	}
+
+	if len(iSets) == 1 {
+		if iSets[0].DeletionTimestamp != nil {
+			return iSets, ErrSetsInDeletionState
+		}
+	} else {
+		if iSets[0].DeletionTimestamp != nil || iSets[1].DeletionTimestamp != nil {
+			return iSets, ErrSetsInDeletionState
+		}
+	}
+
+	switch isType {
+	case InstallerTypeMain:
+		if err := verifyMainInstallerSets(iSets); err != nil {
+			logger.Errorf("%v/%v: failed to verify main sets: %v", i.resourceKind, isType, err)
+			return iSets, err
+		}
+	case InstallerTypePre, InstallerTypePost:
+		if len(iSets) != 1 {
+			logger.Errorf("%v/%v: found multiple sets, expected one", i.resourceKind, isType)
+			return iSets, ErrInvalidState
+		}
+	case InstallerTypeCustom:
```

Comment:
in case of addon/chains we will need to create multiple installerset other than pre post..
so a custom one.. to pass manifest/installer name.. rather than having default